src/Quotations.ts

```diff
@@ -121,20 +115,34 @@ export function extractFromHtml(messageBody: string): ExtractFromHtmlResult {
     if (xmlDocumentCopy.nodeType === NodeTypes.DOCUMENT_NODE && !xmlDocumentCopy.documentElement)
       (xmlDocumentCopy.documentElement as any) = <HTMLElement>xmlDocumentCopy.childNodes[0];
 
+    // Cut empty blockQuote markers
+    cutQuotation(xmlDocumentCopy, {onlyRemoveEmptyBlocks: true});
+
     // Serialize and return.
     return {
       body: xmlDomSerializer.serializeToString(xmlDocumentCopy, true),
       didFindQuote: true
     }
   }
+   // Try and cut the quote of one of the known types.
+   const cutQuotations = cutQuotation(xmlDocumentCopy);
+
   // Otherwise, if we found a known quote earlier, return the content before.
-  else if (!extractQuoteHtml.quoteWasFound && cutQuotations)
+  if (cutQuotations)
     return { body: xmlDomSerializer.serializeToString(xmlDocumentCopy, true), didFindQuote: true };
   // Finally, if no quote was found, return the original HTML.
   else
     return { body: messageBody, didFindQuote: false };
 }
 
+function cutQuotation(xmlDocument: Document, options?: cutQuoteOption) {
```

Comment:
Types should be uppercase.

```suggestion
function cutQuotation(xmlDocument: Document, options?: CutQuoteOption) {
```